packages/payload/src/admin/components/utilities/Preferences/index.tsx

```diff
@@ -65,14 +67,57 @@ export const PreferencesProvider: React.FC<{ children?: React.ReactNode }> = ({
   )
 
   const setPreference = useCallback(
-    async (key: string, value: unknown): Promise<void> => {
-      preferencesRef.current[key] = value
-      await requests.post(
-        `${serverURL}${api}/payload-preferences/${key}`,
-        requestOptions(value, i18n.language),
-      )
+    async (key: string, value: unknown, merge = false): Promise<void> => {
+      if (merge === false) {
+        preferencesRef.current[key] = value
+        await requests.post(
+          `${serverURL}${api}/payload-preferences/${key}`,
+          requestOptions(value, i18n.language),
+        )
+        return
+      }
+
+      let newValue = value
+      const currentPreference = await getPreference(key)
+      if (
+        typeof value === 'object' &&
+        typeof preferencesRef.current[key] === 'object' &&
+        typeof newValue === 'object'
+      ) {
+        newValue = { ...(currentPreference || {}), ...value }
+        if (isDeepEqual(newValue, currentPreference)) {
+          return
+        }
+        pendingUpdate.current[key] = {
+          ...pendingUpdate.current[key],
+          ...(newValue as Record<string, unknown>),
+        }
+      } else {
+        if (newValue === currentPreference) {
+          return
+        }
+        pendingUpdate.current[key] = newValue
+      }
+
+      const updatePreference = async () => {
+        if (isDeepEqual(pendingUpdate.current[key], preferencesRef.current[key])) {
```

Comment:
Could use currentPreference here instead of preferencesRef.current[key]

Reply:
I tried `currentPreference` but it actually needs to compare the most updated value directly on context or it would still send multiple requests here since we aren't reassigning `currentPreference` also.